File: test-suite/golang/guards_clean.go
package inventory

import "fmt"

// CatalogTotal checks the envelope before walking it (early-exit guard).
func CatalogTotal(resp *CatalogResponse) int {
	if resp == nil || resp.Catalog == nil {
		return 0
	}
	return resp.Catalog.Page.Items.Total
}

// ShipmentLabel guards every level it dereferences.
func ShipmentLabel(client *ShipClient) string {
	if client == nil || client.Conn == nil {
		return ""
	}
	return fmt.Sprintf("%s/%s", client.Conn.Region.Host.Name, client.Conn.Region.Code)
}

// WarehouseName guards inside the condition and uses the chain in the body.
func WarehouseName(cfg *InventoryConfig) string {
	if cfg != nil && cfg.Stores != nil {
		return cfg.Stores.Primary.Warehouse.Name
	}
	return "unknown"
}

// SafeName never dereferences: it formats through the nil-safe fallback.
func SafeName(cfg *InventoryConfig) string {
	name := "unknown"
	if cfg != nil && cfg.Stores != nil && cfg.Stores.Primary != nil {
		name = cfg.Stores.Primary.Warehouse.Name
	}
	return name
}

File: test-suite/java/guards_clean.java
package com.example.inventory;

class OrderTotalClean {

    int catalogTotal(Customer customer) {
        if (customer == null || customer.profile == null) {
            return 0;
        }
        return customer.profile.stats.orders.total;
    }

    String shipmentLabel(ShipClient client) {
        if (client instanceof ShipClient) {
            return client.conn.region.host.name;
        }
        return "";
    }

    int safeTotal(Customer customer) {
        return customer != null ? customer.profile.stats.orders.total : 0;
    }

    int requiredTotal(Customer customer) {
        Objects.requireNonNull(customer, "customer");
        if (Objects.isNull(customer.profile)) {
            return 0;
        }
        return customer.profile.stats.orders.total;
    }
}
